### ISSUE TITLE
Download from lightbox fails to authenticate

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#2618]
Steps to reproduce:
* Log into an org on zulipchat.com, or any other server running a recent release and on Xenial. (Not chat.zulip.org, which is on Trusty.)
* Find a conversation with a user-uploaded image in it. (Or just upload one.)
* Tap the image, bringing up the lightbox.
* Tap the ellipsis in the lower right to bring up a menu; select "Download file".
* The download fails. It should succeed.

I tested this just now with 13.4.88 on Android. I'm pretty sure the cause is that we just haven't fixed the download to provide authentication to the server. For more details, see commit cb3198606, where we fixed the display of images within the message list webview.

**Comment by gnprice** [comment]
Relatedly, a user on iOS reported they were unable to open attachments. This comment is a note to myself to file a proper issue for that, which will probably be a followup to this one.

**Comment by borisyankov** [comment]
Also, the issue could be the auth headers which we can not provide with this API - `CameraRoll.saveToCameraRoll` accepts a `string`, not a resource.

**Comment by gnprice** [comment]
Still repro's on 17.1.98 on Android.

**Comment by gnprice** [comment]
@jackrzhang FYI: I just filed #3018 discussing why it'd be good to combine our `.android.js`/`.ios.js` file pairs. I noticed today there seems to be one of them involved here; in a PR addressing this issue it'd probably be helpful to go ahead and combine that pair, up front at the start of the branch.

**Comment by jackrzhang** [comment]
A note relevant to Greg's comment above - 

https://github.com/zulip/zulip-mobile/commit/7eb97b44 was merged, which combined `downloadFile.ios.js` and `downloadFile.android.js`.

**Comment by gnprice** [comment]
@jackrzhang just fixed the iOS side of this, with #3072 !

The codepath where the problem lies is actually totally separate between platforms, so Android will take a separate fix. Just now it's blocked by #3115 , which prevents downloads from working at all on Android; after that we'll come back and complete this.

**Comment by gnprice** [comment]
#3115 was fixed back in November. #3124 now describes the current status; following a fix to a server issue, it looks like authentication may not be the problem anymore. Anyway, closing in favor of the more-specific #3124. (Thanks again @jackrzhang for your work on all the pieces of this!)
